src/replication.c

```diff
@@ -1582,18 +1596,45 @@ int istgt_lu_create_snapshot(spec_t *spec, char *snapname, int io_wait_time, int
 		return false;
 	}
 
+	io_seq = ++spec->io_seq;
+
+	rcommon_mgmt_cmd_t *rmgmt = allocate_rcommon_mgmt_cmd(0);
+	TAILQ_FOREACH(replica, &spec->rq, r_next) {
+		(void) send_replica_snapshot(spec, replica, io_seq, snapname, ZVOL_OPCODE_SNAP_PREPARE, rmgmt);
+	}
+
+	timesdiff(CLOCK_MONOTONIC_COARSE, last, now, diff);
+	MTX_LOCK(&rmgmt->mtx);
+
+	while (diff.tv_sec < wait_time) {
+		if ((rmgmt->cmds_sent == rmgmt->cmds_succeeded) ||
+		    rmgmt->cmds_failed)
+			break;
+		MTX_UNLOCK(&rmgmt->mtx);
+		MTX_UNLOCK(&spec->rq_mtx);
+		sleep(1);
+		MTX_LOCK(&spec->rq_mtx);
+		MTX_LOCK(&rmgmt->mtx);
+		timesdiff(CLOCK_MONOTONIC_COARSE, last, now, diff);
+	}
+	MTX_UNLOCK(&rmgmt->mtx);
+
+	if (rmgmt->cmds_sent != rmgmt->cmds_succeeded) {
+		MTX_UNLOCK(&spec->rq_mtx);
+		free(rmgmt);
```

Comment:
this can't be deleted here as there may be pending responses from replica, and during response handling, this structure will be used.